### PR TITLE
Remove subsets of analysis datasets

### LINKED ISSUE #341: Remove dataset combo box
Users are more likely to get inaccurate fits from choosing different subsets of their data (e.g. forward, backward) and as a result, the dataset combo box in the Analyzer should be removed in addition to any analysis code made to store these different subsets.

For reference this is the combo box:
image

### PR DESCRIPTION
**Pull request by Piphi5**
Fixes #341

### CODE REVIEW

**Comment by calcmogul** [comment]
Looks like some files may be able to have the StringMap include removed.

File: sysid-application/src/main/native/include/sysid/analysis/Storage.h [hunk 1]
```diff
@@ -77,6 +77,8 @@ struct Storage {
    * Dataset for fast (aka dynamic) test
    */
   std::vector<PreparedData> fast;
+
+  bool empty() const { return fast.empty() || slow.empty(); }
```
Comment (calcmogul) [inline sysid-application/src/main/native/include/sysid/analysis/Storage.h:81]: Please make the order of empty() calls match the member declaration order. :)

**Comment by calcmogul** [comment]
The analyzer reset button is overlapping the file select box. Since it resets the whole analyzer, I guess it makes sense in the corner. The select box should just be shrunk a bit.
Screenshot_20220131_213244
Also, the analyzer window height could be reduced in UILayout.h to leave more room for the program log window. Just leave one text box worth of space for the trackwidth box in the angular drivetrain test.

**Comment by Piphi5** [comment]
Here's a sample with elevator data 
image

**Comment by calcmogul** [comment]
Looks like another 5 pixels or so would get rid of the analysis window scrollbar.

**Comment by Piphi5** [comment]
Yeah its gone:
image

**Review by calcmogul — APPROVED** [review]

**Pull request merged**